matrix_functions_types.py

```diff
@@ -39,17 +39,19 @@ class RootInvConfig(MatrixFunctionConfig):
 class EigenConfig(RootInvConfig, EigenvalueDecompositionConfig):
     """Configuration for matrix root inverse via an eigendecomposition.
 
-    Args:
+    Attributes:
         retry_double_precision (bool): Whether to re-trying eigendecomposition with higher (double) precision if lower precision fails due
             to CuSOLVER failure. (Default: True)
         make_positive_semidefinite (bool): Perturbs matrix eigenvalues to ensure it is numerically positive semi-definite. (Default: True)
         exponent_multiplier (float): Number to be multiplied to the numerator of the inverse root, i.e., eta where the
             exponent is -eta / (2 * p). (Default: 1.0)
+        eigen_decomp_offload_device (str): Device to offload eigen decomposition to. If value is empty string, we don't perform offloading. (Default: "")
```

Comment:
This could also be moved into `EigenvalueDecompositionConfig` instead, such that `EighEigenvectorConfig` can also use it to compute the eigenvectors (after the corresponding changes in `matrix_functions.py`).

Reply:
So we don't want to support the same functionality for other matrix functions that also use the `eigh` eigendecomposition?